pages/posts/[boardType]/index.tsx

```diff
@@ -0,0 +1,52 @@
+import { useRouter } from "next/router";
+import { useQuery } from "react-query";
+import { api } from "../../../api/api";
+import PageTitle from "../../../components/PageTitle";
+import {
+  boardList,
+  PostsBoardNavbar,
+} from "../../../components/PostsBoardNavbar";
+import SignInForm from "../../../components/SignInForm";
+import { useAppSelector } from "../../../store/hooks";
+import { IPostsData } from "../../../types/types";
+import { PostsList } from "../PostsList";
+
+export default function PostsBoard() {
+  const router = useRouter();
+  const { boardType } = router.query;
+
+  const boardId = boardList.find((board) => board.route === boardType)?.id;
+
+  const token = useAppSelector((state) => state.auth.token);
+
+  const { data, status, error } = useQuery(
+    ["posts", boardType],
+    async () => {
+      const response: IPostsData = await api.get(`/posts?boardId=${boardId}`);
+      return response;
+    },
```

Comment:
크게 중요한 건 아니지만.. 이렇게 쓰는 건 어떨까요? 👀 

```suggestion
    () => api.get<IPostsData>(`/posts?boardId=${boardId}`),
```